### DISCUSSION TITLE
Most recent action card

### DISCUSSION THREAD

**Discussion by Winniepoom** [Budibase/budibase#11565] [Help] [answered]
Hello Budibase team,

I would like to implement a recent event card that displays the most recent actions that have been taken. I have already set up an employee table and a right panel to assign jobs to employees. Now, when I assign a client job to John, I want the recent event card to display a message like: 'John has been assigned a client job' or something similar .

Thank you.

**Comment by ConorWebb96** [comment]
Hey @Winniepoom,

Theres a few options you could do here. You could create a table called `event history` create all the columns which you'd require. Then use the first row of this event history table for the event card. (This would require you to chain saving actions to it when you update other records within your app)

The other easier option is to create a state object with the `update state` action and add all the relevant information to this object. From here you can bind it up to your event card to display the information using the state object you created. (You can persist it as well so it will show after a refresh.)
https://docs.budibase.com/docs/app-state 

I hope one of these options point you in the right direction.

**Comment by Winniepoom** [comment]
@ConorWebb96 
You're pointing me in the right direction, but there seems to be a slight issue. Even after updating the state with the save button in the right panel and adding information to the object, the recent action card still displays the old value. For example, if Mick was initially assigned to Project 1 and then I reassigned him to Project 2 and hit save, the card still updates to "Mick was assigned to Project 1." Subsequently, when I change Mick's assignment to Project 3, the card continues to display "Mick was assigned to Project 2." The problem persists with the card not reflecting the updated values.

I'm unsure whether this issue originates from my database or not, as it takes approximately 6 seconds for my table to update after making changes in the foam.

![tempsnip](https://github.com/Budibase/budibase/assets/107398226/611a00b5-cd05-4f43-a6fb-83c94a523ac4)

Here my action 
![image](https://github.com/Budibase/budibase/assets/107398226/cab92f94-f7d5-4d30-a483-19f365d8d767)
![image](https://github.com/Budibase/budibase/assets/107398226/2fe861df-9ce9-4ac4-9dcf-55ff023666e8)

**Reply by ConorWebb96** [reply, marked as answer]
Hey @Winniepoom,

Instead of using your repeater, why don't you use the form `field values` for the nick_name and project?

This would look something like; `{{ New Form.Fields.nick_name }}`, `{{ New Form.Fields.project }}`. Basically what this will do is use the selected values of your input fields instead of whatever currently is stored in your db.

I hope this helps! 🚀

**Reply by Winniepoom** [reply]
@ConorWebb96 

Thank you, it's working.
FYI, initially, when I tried to use the form field values, I accidentally used the wrong form. Instead of using the right panel form, I used a different one. Haha!

**Reply by ConorWebb96** [reply]
No worries! I'm glad it is working for you now! 🤝

**Comment by Winniepoom** [comment]
@ConorWebb96 
I had more question relate to this.
So if  I want to make a 3 recent card instead of 1 should I have to create a table method to collect event or any other way that more efficiency ? 
Because I can't figure out how to achieve this with state method.
